vllm/forward_context.py

```diff
@@ -84,6 +86,34 @@ def num_tokens_across_dp(num_tokens: int, dp_size: int,
         dist.all_reduce(num_tokens_tensor, group=get_dp_group().cpu_group)
         return num_tokens_tensor
 
+    @staticmethod
+    def should_ubatch_across_dp(should_ubatch: bool, orig_num_tokens_per_ubatch: int,
```

Comment:
Should this really live inside forward_context? Seems like something for one of the ubatch files

Reply:
I'll also add a comment here explaining what I just posted.